packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx

```diff
@@ -90,7 +134,9 @@ const PickersLayout = function PickersLayout<
   const { toolbar, content, tabs, actionBar, shortcuts } = usePickerLayout(props);
   const { sx, className, isLandscape, ref, wrapperVariant } = props;
 
-  const ownerState = props;
+  const theme = useTheme();
+  const isRTL = theme.direction === 'rtl';
+  const ownerState = { ...props, isRTL };
```

Comment:
Do we have a way in the variants to access the theme? Or do we have a way to style depending on the current direction (like we have `applyDarkStyles` for the current mode)?

Reply:
We are discussing adding support for direction similar as for dark theme

Reply:
If I access the theme from the top level styled callback and keep using `theme.direction === 'rtl' ? 3 : 1` it will work with zero runtime?

Reply:
But if I am accessing it from the top level styled callback and I do something like:

```tsx
styled('div')(({ theme }) => ({
  variants: [true, false].map(isLandscape => ({
    props: { isLandscape },
    style: {
      [`& .${pickersLayoutClasses.toolbar}`]: isLandscape
        ? {
            gridColumn: theme.direction === 'rtl' ? 3 : 1,
            gridRow: '2 / 3',
          }
        : { gridColumn: '2 / 4', gridRow: 1 },
      [`.${pickersLayoutClasses.shortcuts}`]: isLandscape
        ? { gridColumn: '2 / 4', gridRow: 1 }
        : {
            gridColumn: theme.direction === 'rtl' ? 3 : 1,
            gridRow: '2 / 3',
          },
    }
  })
}))
```

It will still work with zero runtime?

Reply:
cc @brijeshb42

Reply:
The whole thing will work except the theme.direction part. It will take whatever the value is at build time. So in a way it'll be static to rtl or ltr. You should consider accepting rtl value from ownerState that comes from some context . Or instead @siriwatknp @mnajdova we should prioritise adding a direction helper here.

Reply:
OK, that's very clear for the build time aspect!

> You should consider accepting rtl value from ownerState that comes from some context 

But do we agree that this context should be a feature built by the core team and shared across all components?
We shouldn't start adding custom context for this kind of things on the pickers, on the grid, etc...